src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -240,25 +340,108 @@ object SRAM {
+    */
+  def masked[T <: Data](
+    size:                BigInt,
+    tpe:                 T,
+    readPortClocks:      Seq[Clock],
+    writePortClocks:     Seq[Clock],
+    readwritePortClocks: Seq[Clock]
   )(
-    implicit sourceInfo: SourceInfo
+    implicit evidence: T <:< Vec[_],
+    sourceInfo:        SourceInfo
+  ): SRAMInterface[T] =
+    memInterface_impl(
+      size,
+      tpe
+    )(
+      readPortClocks,
+      writePortClocks,
+      readwritePortClocks,
+      None
+    )(
+      Some(evidence),
+      sourceInfo
+    )
+
+  /** Generates a [[SyncReadMem]] within the current module, connected to an explicit number
+    * of read, write, and read/write ports, with masking capability on all write and read/write ports.
+    * Each port is clocked with its own explicit `Clock`, rather than being given the implicit clock.
+    *
+    * @param size The desired size of the inner `SyncReadMem`
+    * @tparam T The data type of the memory element
+    * @param readPortClocks A sequence of clocks for each read port; and (numReadPorts + numReadwritePorts) > 0
+    * @param writePortClocks A sequence of clocks for each write port; and (numWritePorts + numReadwritePorts) > 0
+    * @param readwritePortClocks A sequence of clocks for each read-write port; and the above two conditions must hold
+    * @param memoryFile A memory file whose path is emitted as Verilog directives to initialize the inner `SyncReadMem`
+    *
+    * @return A new `SRAMInterface` wire containing the control signals for each instantiated port
+    * @note The size of each `Clock` sequence determines the corresponding number of read, write, and read-write ports
+    * @note This does *not* return the `SyncReadMem` itself, you must interact with it using the returned bundle
+    * @note Read-only memories (R >= 1, W === 0, RW === 0) and write-only memories (R === 0, W >= 1, RW === 0) are not supported by this API, and will result in an error if declared.
+    */
+  def masked[T <: Data](
+    size:                BigInt,
+    tpe:                 T,
+    readPortClocks:      Seq[Clock],
+    writePortClocks:     Seq[Clock],
+    readwritePortClocks: Seq[Clock],
+    memoryFile:          MemoryFile
+  )(
+    implicit evidence: T <:< Vec[_],
+    sourceInfo:        SourceInfo
+  ): SRAMInterface[T] =
+    memInterface_impl(
+      size,
+      tpe
+    )(
+      readPortClocks,
+      writePortClocks,
+      readwritePortClocks,
+      Some(memoryFile)
+    )(
+      Some(evidence),
+      sourceInfo
+    )
+
+  private def memInterface_impl[T <: Data](
+    size:                BigInt,
+    tpe:                 T
+  )(readPortClocks:      Seq[Clock],
+    writePortClocks:     Seq[Clock],
+    readwritePortClocks: Seq[Clock],
+    memoryFile:          Option[MemoryFile]
+  )(evidenceOpt:         Option[T <:< Vec[_]],
+    sourceInfo:          SourceInfo
```

Comment:
Is there a real purpose to having 3 parameter lists here? Since this is a private method I would just defined it with a normal single parameter list.
```suggestion
  private def memInterface_impl[T <: Data](
    size:                BigInt,
    tpe:                 T,
    readPortClocks:      Seq[Clock],
    writePortClocks:     Seq[Clock],
    readwritePortClocks: Seq[Clock],
    memoryFile:          Option[MemoryFile],
    evidenceOpt:         Option[T <:< Vec[_]],
    sourceInfo:          SourceInfo
```